### PR TITLE
[ci] Build CW310 hyperdebug bitstream for master

### PR DESCRIPTION
**Pull request by a-will**
Add job to build the CW310 hyperdebug bitstream for merged PRs, and
include the hyperdebug bitstream in uploads to the bitstream cache. Use
the fusesoc core name for a subdirectory in which to place the bitstream
and MMI files.

### CODE REVIEW

File: azure-pipelines.yml
Comment (a-will) [inline azure-pipelines.yml:453]: The condition of doing a run for the "master" branch is removed for testing. It needs to be added back before merging.

File: azure-pipelines.yml
Comment (a-will) [inline azure-pipelines.yml:536]: Ah, need to make sure to download and unpack these.

File: azure-pipelines.yml
Comment (milesdai) [inline azure-pipelines.yml:510]: Were you able to test if this download succeeds if the hyperdebug job did not run, or do we also need the `${{ if eq(variables['Build.SourceBranchName'], 'master') }}` construct used below on line 526?

**Review by milesdai — APPROVED** [review]
LGTM modulo a few questions

File: azure-pipelines.yml
Comment (a-will) [reply azure-pipelines.yml:483]: Yup, exactly

Comment (a-will) [reply azure-pipelines.yml:483]: Well, actually, it's to publish it for the splicing job, in addition to debugging. We need to send one artifact up to the bitstreams cache, and that is done in the splicing job.

File: azure-pipelines.yml
Comment (a-will) [reply azure-pipelines.yml:510]: Not yet, but it'll get tested once I add the condition for running the job back ;)

You're probably right, though. I'll add the conditional inclusion of that job's results.

**Review by timothytrippel — APPROVED** [review]

**Comment by a-will** [comment]
Does anyone have any idea why the CW310 Tests job was skipped? I am confused!

**Comment by a-will** [comment]
> Does anyone have any idea why the CW310 Tests job was skipped? I am confused!

It would appear that the default condition is to traverse the entire dependency graph for the job and require that every node be 'Succeeded' or 'SucceededWithIssues'. That means that an internal node that is 'Skipped' will fail the condition.

I'm double-checking this by specifying a succeeded() condition against just the neighbors, but if true, it would be kind of an odd default.

**Comment by a-will** [comment]
Merging, as the failing tests are known problems.

**Pull request merged**
